var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -837,6 +879,7 @@ def cmake_args(self):
             define("LLVM_REQUIRES_RTTI", True),
             define("LLVM_ENABLE_RTTI", True),
             define("LLVM_ENABLE_LIBXML2", False),
+            define("LLVM_ENABLE_ASSERTIONS", True),
```

Comment:
This is not a good default

Reply:
Can you elaborate on why this might not be a good default?

Reply:
That mode is explicitly intended for LLVM developers, and is enabled by default when building in Debug mode and disabled in all other build types.  If someone wants that, they really should be enabling debug IMO.